zanata-war/src/main/java/org/zanata/action/LanguageJoinAction.java

```diff
@@ -97,65 +92,164 @@
+        return getRequestedRole(request);
     }
 
-    public void bindRole(String role, boolean checked) {
-        if (role.equals("translator")) {
-            requestAsTranslator = checked;
-        } else if (role.equals("reviewer")) {
-            requestAsReviewer = checked;
-        } else if (role.equals("coordinator")) {
-            requestAsCoordinator = checked;
-        }
+    public String getRequestedRole(LanguageRequest request) {
+        return Joiner.on(", ")
+            .skipNulls()
+            .join(request.getCoordinator() ? "Coordinator" : null,
+                request.getReviewer() ? "Reviewer" : null,
+                request.getTranslator() ? "Translator" : null);
+    }
+
+    public void acceptRequest(Long languageRequestId) {
+        identity.checkPermission(getLocale(), "manage-language-team");
+        LanguageRequest request =
+                requestServiceImpl.getLanguageRequest(languageRequestId);
+
+        languageTeamServiceImpl.joinOrUpdateRoleInLanguageTeam(
+                language, request.getRequest().getRequester().getId(),
+                request.getTranslator(), request.getReviewer(),
+                request.getCoordinator());
+
+        requestServiceImpl.updateLanguageRequest(languageRequestId,
+                authenticatedAccount, RequestState.ACCEPTED, "");
+        facesMessages.addGlobal(msgs.get("jsf.language.request.updated"));
+    }
+
+    public void declineRequest(Long languageRequestId, String comment) {
+        identity.checkPermission(locale, "manage-language-team");
+        requestServiceImpl.updateLanguageRequest(languageRequestId,
+            authenticatedAccount, RequestState.REJECTED, comment);
+        facesMessages.addGlobal(msgs.get("jsf.language.request.updated"));
     }
 
-    private void reset() {
-        requestAsTranslator = false;
-        requestAsReviewer = false;
-        requestAsCoordinator = false;
-        message = "";
+    @CheckLoggedIn
+    public void requestAsTranslator() {
+        processRequest(true, false, false);
     }
 
     @CheckLoggedIn
-    public void send() {
+    public void requestAsReviewer() {
+        processRequest(false, true, false);
+    }
+
+    @CheckLoggedIn
+    public void requestAsCoordinator() {
+        processRequest(false, false, true);
+    }
+
+    private void processRequest(boolean isRequestAsTranslator,
+            boolean isRequestAsReviewer,
+            boolean isRequestAsCoordinator) {
+        try {
+            requestServiceImpl
+                    .createLanguageRequest(authenticatedAccount, getLocale(),
+                            isRequestAsCoordinator,
+                            isRequestAsReviewer, isRequestAsTranslator);
+            sendRequestEmail(isRequestAsCoordinator, isRequestAsReviewer,
+                isRequestAsTranslator);
+        } catch (RequestExistException e) {
+            log.warn("Request already exist for {0} in language {1}.",
```

Comment:
"exist" should be "exists".  But shouldn't we tell the user about this problem, instead of logging a warning?